tensorlayer/models/seq2seq.py

```diff
@@ -0,0 +1,154 @@
+#! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+import tensorflow as tf
+import tensorlayer as tl
+import numpy as np
+from tensorlayer.models import Model
+from tensorlayer.layers import Dense, Dropout, Input
+from tensorlayer.layers.core import Layer
+
+
+class Seq2seq(Model):
+
+    """vanilla stacked layer Seq2Seq model.
+
+    Parameters
+    ----------
+    decoder_seq_length: int
+        The length of your target sequence
+    cell_enc : str, tf.function
+        The RNN function cell for your encoder stack, i.e. tf.keras.layers.GRUCell
+    cell_dec : str, tf.function
+        The RNN function cell for your decoder stack, i.e. tf.keras.layers.GRUCell
+    n_layer : int
+        The number of your RNN layers for both encoder and decoder block
+    embbedding_layer : tl.function
+        The embedding layer function, i.e. tl.layers.Embedding(vocabulary_size=voc_size, embedding_size=emb_dim)
+    is_train : bool
+        True if train mode, False if Inference mode
+    name : str
+        The model name
+    
+    Examples
+    ---------
+    Classify stacked-layer Seq2Seq model, see `chatbot <https://github.com/tensorlayer/seq2seq-chatbot>`__
+
+    Returns
+    -------
+        static stacked-layer Seq2Seq model.
+    """
+    def __init__(
+            self,
+            decoder_seq_length,
+            cell_enc,
+            cell_dec,
+            n_units=256,
+            n_layer=3,
+            embedding_layer=None,
+            is_train=True,
+            name="seq2seq_"
```

Comment:
`name=None` by default. There is an auto-naming mechanism in `tl.models.Model()`.